src/main/java/dev/j3fftw/litexpansion/items/WateringCan.java

```diff
@@ -0,0 +1,209 @@
+
+                            if (!updateUses(p, item, 1))
+                                return;
+                            blockLocation.getWorld().spawnParticle(Particle.WATER_SPLASH, blockLocation, 0);
+                            double random = Math.random();
+                            if (random <= Constants.SUGAR_CANE_WATERING_SUCCESS_CHANCE) {
+                                above.setType(Material.SUGAR_CANE);
+                                blockLocation.getWorld().playEffect(blockLocation, Effect.VILLAGER_PLANT_GROW, 0);
+                            }
+
+                            } else {
+                            Utils.send(p, "&cThe sugar cane is obstructed!");
+                        }
+
+                    } else if (blockData instanceof Ageable) {
+
+                        Ageable crop = (Ageable) blockData;
+                        int currentAge = crop.getAge();
+                        int maxAge = crop.getMaximumAge();
+
+                        if (currentAge < maxAge && updateUses(p, item, 1)) {
+                            blockLocation.getWorld().spawnParticle(Particle.WATER_SPLASH, blockLocation, 0);
+                            double random = Math.random();
+                            if (random <= Constants.CROP_WATERING_SUCCESS_CHANCE) {
+                                crop.setAge(currentAge + 1);
+                                blockLocation.getWorld().playEffect(blockLocation, Effect.VILLAGER_PLANT_GROW, 0);
+                            }
+
+                        } else {
+                            Utils.send(p, "&cThis crop is already ready for harvest!");
+                            return;
+                        }
+
+                        b.setBlockData(blockData);
+                    } else if (Tag.SAPLINGS.isTagged(b.getType())) {
+
+                        if (BlockStorage.check(b) != null) {
+                            Utils.send(p, "&cSorry, this is a Slimefun plant!");
+
+                        } else {
+
+                            if (!updateUses(p, item, 1))
+                                return;
+                            blockLocation.getWorld().spawnParticle(Particle.WATER_SPLASH, blockLocation, 0);
+                            double random = Math.random();
+                            if (random <= Constants.TREE_WATERING_SUCCESS_CHANCE) {
+
+                                TreeType treeType = TreeType.TREE;
+                                String parseSapling = b.getType().toString()
+                                    .replace("_SAPLING", "");
+
+                                if (!parseSapling.equals("OAK")) {
+                                    if (parseSapling.equals("JUNGLE")) {
+                                        parseSapling = "SMALL_JUNGLE";
+                                    }
+                                    treeType = TreeType.valueOf(parseSapling);
+                                }
```

Comment:
This should probably become its own method really. Like `getTreeTypeFromSapling`